src/main/java/gregtech/client/model/modelfactories/ActiveVariantBlockBakedModel.java

```diff
@@ -0,0 +1,101 @@
+package gregtech.client.model.modelfactories;
+
+import gregtech.api.block.VariantActiveBlock;
+import gregtech.client.utils.BloomEffectUtil;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.block.model.ItemOverrideList;
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.client.MinecraftForgeClient;
+import net.minecraftforge.common.property.IExtendedBlockState;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static gregtech.common.blocks.MetaBlocks.statePropertiesToString;
+
+public class ActiveVariantBlockBakedModel implements IBakedModel {
+    private final ThreadLocal<TextureAtlasSprite> particle;
+
+    ActiveVariantBlockBakedModel() {
+        this.particle = ThreadLocal.withInitial(() -> Minecraft.getMinecraft().getTextureMapBlocks().getMissingSprite());
+    }
+
+    @Override
+    public List<BakedQuad> getQuads(@Nullable IBlockState state, @Nullable EnumFacing side, long rand) {
+        List<BakedQuad> quads = Collections.emptyList();
```

Comment:
Could this just be `List<BakedQuad> quads = new ArrayList<>();` so that you don't have to redeclare `quads` on line 47?